price_feed_poller.py

```diff
@@ -20,75 +18,101 @@ def handle_requestUpdate():
 
     print(f"Got {balance} wei")
 
-
-    dr_id = pricefeed.functions.requestUpdate().transact(
+    # Hardcoded gas since it does not estimate well
+    dr_id = pricefeedcontract.functions.requestUpdate().transact(
         {"from": account_addr, "gas": 310000, "value": 310000000000000})
-    post_tx_hash = bytes(dr_id)
-    w3.eth.waitForTransactionReceipt(dr_id)
-    requestId = pricefeed.functions.lastRequestId().call()
 
-    print(
+     # Get receipt of the transaction   
+    receipt = w3.eth.waitForTransactionReceipt(dr_id)
```

Comment:
According to the documentation, `waitForTransactionReceipt` raises an exception if the transaction is not included after the default timeout of 120 seconds.

How do we handle that case? Wait or retry with a higher gas price?

https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.waitForTransactionReceipt